### PR TITLE
Module 14

### PR DESCRIPTION
**Pull request by sergei820**
Hello! The task is implemented, please, check

module_14_done py

### CODE REVIEW

File: module_14/tests/demoblaze_test.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+@pytest.mark.usefixtures("setup")
+class TestDemoBlaze:
+
+    def test_login(self):
+        home_page = HomePage(self.driver)
+        home_page.open_home_page()
+        home_page.click_login_button()
+        home_page.log_in("snowfallslow@gmail.com", "1qa2ws3ed")
```
Comment (AnastasiiaGor) [inline module_14/tests/demoblaze_test.py]: It's better avoid hard-code like this and keep data as constants separately

File: module_14/tests/demoblaze_test.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        home_page.check_log_out_button_is_visible()
+        home_page.validate_welcome_message_text("snowfallslow@gmail.com")
+
+    def test_add_to_cart(self):
```
Comment (AnastasiiaGor) [inline module_14/tests/demoblaze_test.py]: this test has a specific precondition User is logged in that was intended to be a fixture

File: module_14/pages/home_page.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        assert ec.visibility_of_element_located(self.USERNAME_INPUT)
+
+    def validate_welcome_message_text(self, username: str):
+        self.wait.until(ec.visibility_of_element_located(self.WELCOME_MESSAGE))
+        welcome_message = self.driver.find_element(*self.WELCOME_MESSAGE).text
+        assert welcome_message == f"Welcome {username}"
+
+    def click_category(self, category):
+        self.driver.refresh()
+        category_selector = "//div[@class='list-group']/a[text()='CATEGORY']".replace('CATEGORY', category)
```
Comment (AnastasiiaGor) [inline module_14/pages/home_page.py]: https://www.w3schools.com/python/ref_string_format.asp format method could be used instead

**Review by AnastasiiaGor — CHANGES_REQUESTED** [review]

File: module_14/tests/demoblaze_test.py (hunk 1 above)
Comment (sergei820) [reply module_14/tests/demoblaze_test.py]: Thank you! Fixed

File: module_14/pages/home_page.py (hunk 3 above)
Comment (sergei820) [reply module_14/pages/home_page.py]: Oh, .format() method is nice! It turned out that it even works just with f"... {category} ...", can I keep it like on the screenshot below? It looks even shorter:
image

File: module_14/tests/demoblaze_test.py (hunk 1 above)
Comment (sergei820) [reply module_14/tests/demoblaze_test.py]: Thank you for this point! Fixed.

File: module_14/tests/demoblaze_test.py (hunk 2 above)
Comment (sergei820) [reply module_14/tests/demoblaze_test.py]: Thank you for this comment, Anastasiia! Now I understand how to pass some object from one fixture to another:)

**Review by AnastasiiaGor — APPROVED** [review]

**Pull request merged**